src/main/java/com/google/jenkins/plugins/storage/AbstractUpload.java

```diff
@@ -540,7 +540,11 @@ public void act() throws IOException, InterruptedException, ExecutorException {
                 new Predicate<ObjectAccessControl>() {
                   @Override
                   public boolean apply(ObjectAccessControl access) {
-                    return Objects.equal(access.getEntity(), publicEntity);
+                    if (access != null) {
+                      return Objects.equal(access.getEntity(), publicEntity);
+                    } else {
+                      throw new NullPointerException(Messages.AbstractUpload_NullAccessError());
```

Comment:
I know this is not a convenient place to add the listener/logger, but I'm not sure about throwing an exception on purpose here. Because this just returns a boolean, would it make sense to just log and return false, or is there somewhere else that depends on the exception throwing behavior?

Reply:
Hmm yea, I wasn't too sure about this case either. I'll have to see what are the consequences of just returning a boolean (which is what I originally wanted to do, but I was worried it was hacky).

This was mainly to fulfill a checkstyle requirement.

Reply:
+1 If throwing a NPE is the desired behavior then suggested approach is with Precondition.checkNotNull

Reply:
I tried Precondition.checkNull previously, but that wasn't enough to satisfy the checkStyle requirement. 

Looking at the code, the iterable is an empty list if the access control is null. I'm inclined to throw an exception because that is highly unexpected behavior?

Reference:
https://github.com/jenkinsci/google-storage-plugin/blob/631adc9c0724bd404012277cfe9e38087d8eb40f/src/main/java/com/google/jenkins/plugins/storage/AbstractUpload.java#L523-L526

I can still add a Precondition.checkNotNull, but looks like checkStyle still wants me to do explicitly do a null check.

Reply:
I would suggest removing the maven-checkstyle-plugin entirely from the project. A lot of the cases it checks for are already handled by the fmt-maven-plugin.

Reply:
My mistake, I meant spotBugs. What do you propose we do in this situation? 
I tried only doing Preconditions.checkNotNull and that didn't pass, and it also complains of redundancy when added and with a null check.

I threw an exception since getDefaultObjectAcl already does a null check. bucket.getDefaultObjectAcl either returns some value or null for none.